src/ert/parsing/config_schema_item.py

```diff
@@ -0,0 +1,290 @@
+                    ).set_context(token)
+                )
+
+            if not os.access(absolute_path, os.X_OK):
+                context = (
+                    f"{token.value!r} which was resolved to {absolute_path!r}"
+                    if token.value != absolute_path
+                    else f"{token.value!r}"
+                )
+                raise ConfigValidationError.from_info(
+                    ErrorInfo(
+                        message=f"File not executable: {context}",
+                        filename=token.filename,
+                    ).set_context(token)
+                )
+            return ContextString(absolute_path, token, keyword)
+        return ContextString(str(token), token, keyword)
+
+    def apply_constraints(
+        self,
+        args: List[Any],
+        keyword: FileContextToken,
+        cwd: str,
+    ) -> Union[List[Any], Any]:
+        errors: List[Union[ErrorInfo, ConfigValidationError]] = []
+
+        args_with_context = []
+        for i, x in enumerate(args):
+            if isinstance(x, FileContextToken):
+                try:
+                    value_with_context = self.token_to_value_with_context(
+                        x, i, keyword, cwd
+                    )
+                    args_with_context.append(value_with_context)
+                except ConfigValidationError as err:
+                    errors.append(err)
+                    continue
+            else:
+                args_with_context.append(x)
+
+        if self.argc_min != -1 and len(args) < self.argc_min:
+            errors.append(
+                ErrorInfo(
+                    message=f"{self.kw} must have at least {self.argc_min} arguments",
+                    filename=keyword.filename,
+                ).set_context(ContextString.from_token(keyword))
+            )
+        elif self.argc_max != -1 and len(args) > self.argc_max:
+            errors.append(
+                ErrorInfo(
+                    message=f"{self.kw} must have maximum {self.argc_max} arguments",
+                    filename=keyword.filename,
+                ).set_context(ContextString.from_token(keyword))
+            )
+
+        if len(errors) > 0:
+            raise ConfigValidationError.from_collected(errors)
+
+        if self.argc_max == 1 and self.argc_min == 1:
+            return args_with_context[0]
+
+        return args_with_context
+
+    def join_args(self, line: List[Any]) -> List[Any]:
+        n = self.join_after
+        if 0 < n < len(line):
+            joined = FileContextToken.join_tokens(line[n:], " ")
+            new_line = line[0:n]
+            if len(joined) > 0:
+                new_line.append(joined)
+            return new_line
+        return line
+
+
+def check_required(
+    schema: Mapping[str, SchemaItem],
+    declared_kws: Set[str],
+    filename: str,
+) -> None:
+    errors: List[ErrorInfo] = []
+
+    # schema.values()
+    # can return duplicate values due to aliases
```

Comment:
This function seems to be duplicated now.

Reply:
Done, removed entire check_required method in `config_schema_item.py`